src/openms/include/OpenMS/ANALYSIS/TOPDOWN/FLASHDeconvAlgorithm.h

```diff
@@ -0,0 +1,104 @@
+// For a full list of authors, refer to the file AUTHORS.
+// --------------------------------------------------------------------------
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Kyowon Jeong, Jihyung Kim $
+// $Authors: Kyowon Jeong, Jihyung Kim $
+// --------------------------------------------------------------------------
+
+#pragma once
+
+#include <OpenMS/APPLICATIONS/TOPPBase.h>
+#include <OpenMS/CONCEPT/Types.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/IsotopeDistribution.h>
+#include <OpenMS/CHEMISTRY/ISOTOPEDISTRIBUTION/CoarseIsotopePatternGenerator.h>
+#include <OpenMS/MATH/STATISTICS/StatisticFunctions.h>
+#include <OpenMS/KERNEL/MSSpectrum.h>
+#include <OpenMS/KERNEL/MSExperiment.h>
+#include <iostream>
+#include <iomanip>
+#include <chrono>
+
+#include <OpenMS/ANALYSIS/TOPDOWN/FLASHDeconvHelperStructs.h>
+#include <OpenMS/ANALYSIS/TOPDOWN/PeakGroup.h>
+#include <OpenMS/ANALYSIS/TOPDOWN/DeconvolutedSpectrum.h>
+#include <OpenMS/ANALYSIS/TOPDOWN/SpectrumDeconvolution.h>
+
+namespace OpenMS
+{
+  /** NEED to be modified
+  @brief @ref
+
+  @htmlinclude
+
+  @ingroup Topdown
+*/
+  /** check OPENMS c++ guide **/
+
+  struct DeconvolutedSpectrum;
+
+  class OPENMS_DLLAPI FLASHDeconvAlgorithm
+  {
+  public:
+    typedef FLASHDeconvHelperStructs::Parameter Parameter;
+    typedef FLASHDeconvHelperStructs::PrecalculatedAveragine PrecalculatedAveragine;
+    typedef FLASHDeconvHelperStructs::LogMzPeak LogMzPeak;
+
+    /// default constructor
+    /// 
+    FLASHDeconvAlgorithm(FLASHDeconvHelperStructs::PrecalculatedAveragine &avg, Parameter &param);
+
+    /// default destructor
+    ~FLASHDeconvAlgorithm();
+
+    /// copy constructor
+    // FLASHDeconvAlgorithm(const FLASHDeconvAlgorithm &);
+
+    /// assignment operator
+    FLASHDeconvAlgorithm &operator=(const FLASHDeconvAlgorithm &fd);
+
+    // deconvolution main function
+    void getPeakGroups(DeconvolutedSpectrum &spec, int &specIndex, int &massIndex);
+
+    // convert double to nominal mass
+    static int getNominalMass(double m);
+
+  protected:
+    // FLASHDeconv parameter set
+    Parameter &param;
```

Comment:
inherit from DefaultParamHandler?

Reply:
would need to take a closer look how this plays with parameter passing from C'

Reply:
Should FLASHDeconvAlgorithm inherit from DefaultParamHandler? There are so many variables specific to FLASHDeconv so I made a separate structure for it. Is it not a good idea?